docs/sdkx-react-native/delegates.mdx

```diff
@@ -0,0 +1,227 @@
+```
+import {
+  helpshiftEmitter,
+} from 'helpshift-plugin-sdkx-react-native';
+
+```
+
+```
+const handleCallbacks = () => {
+    helpshiftEmitter.addListener('onEventOccurred', (data) => {
+      if (data.eventName == 'receivedUnreadMessageCount') {
+        try {
+          let countData =
+            Platform.OS == 'ios' ? data.eventData : JSON.parse(data.eventData);
+          console.log(countData.count);
+        } catch (err) {
+          console.error('Parsing Error', err);
+        }
+      } else {
+        console.log(JSON.stringify(data));
+      }
+    });
+    helpshiftEmitter.addListener('onUserAuthenticationFailure', (data) => {
+      Toast.show({
+        type: 'error',
+        text1: JSON.stringify(data),
+      });
+    });
+  };
+```
+
+
+## Events {#helpshift-events}
+
+### Helpshift session started event {#session-delegates}
+
+This event gets fired when the Helpshift session starts
+
+```
+    helpshiftEmitter.addListener('onEventOccurred', (data) => {
+      if (data.eventName == 'helpshiftSessionStarted') {
+        //Your code here.
+      }
+    });  
+```
+
+### Helpshift session ended event {#session-delegates}
+
+This event gets fired when the Helpshift session ends.
+
+```
+    helpshiftEmitter.addListener('onEventOccurred', (data) => {
+      if (data.eventName == 'helpshiftSessionEnded') {
+        //Your code here.
+      }
+    });  
+```
+
+### Unread Message Count Event {#unreadMessageCount}
+
+This event gets fired when you call `RequestUnreadMessageCount(FetchFromServer)` API to request unread messages count.
+
+```
+    helpshiftEmitter.addListener('onEventOccurred', (data) => {
+      if (data.eventName == 'receivedUnreadMessageCount') {
+        //Your code here.
+      } 
+    });
+    helpshiftEmitter.addListener('onUserAuthenticationFailure', (data) => {
+       //Your code here.
+    });
+```
+### Conversation Status Event {#conversation-status}
+
+This event contains information about the current ongoing conversation.
+
+```
+    helpshiftEmitter.addListener('onEventOccurred', (data) => {
+      if (data.eventName == 'conversationStatus') {
+        //Your code here.
+        //Event data: `latestIssueId`, `latestIssuePublishId`, `open`
+      }
+    });  
+```
+### Widget Toggle Event {#widget-toggle-event}
+
+This event is triggered when the user opens or exits the chat screen. This event is triggered with a boolean value of `"visible"` key. For your reference, see the below example:
+
+```
+    helpshiftEmitter.addListener('onEventOccurred', (data) => {
+      if (data.eventName == 'widgetToggle') {
+        //Your code here.
+      }
+    });  
+```
+### Conversation Start Event {#conversation-start-event}
+
+This event triggers when the user sends the first message in a conversation. The event data object has a key, `message`, which includes the message string the end-user sent to start the conversation. For your reference, see the below example.
```

Comment:
Show in code sample how to read and use "message" from "data"

Reply:
Updated